frontend/packages/dev-console/src/components/import/pipeline/pipeline-template-utils.ts

```diff
@@ -24,28 +29,34 @@ export const createPipelineForImportFlow = async (formData: GitImportFormData) =
     git,
     pipeline,
   } = formData;
-  const template = _.cloneDeep(pipeline.template);
+  const template = _.cloneDeep(pipeline.template) as Pipeline;
 
   template.metadata = {
     name: `${name}`,
     namespace,
     labels: { ...template.metadata.labels, 'app.kubernetes.io/instance': name },
   };
 
-  template.spec.params =
-    template.spec.params &&
-    template.spec.params.map((param) => {
-      if (param.name === 'APP_NAME') {
-        param.default = name;
-      }
-      return param;
-    });
-
-  try {
+  template.spec.params = template.spec.params?.map((param) => {
+    switch (param.name) {
+      case 'APP_NAME':
+        return { ...param, default: name };
+      case 'GIT_REPO':
+        return { ...param, default: git.url };
+      case 'GIT_REVISION':
+        return { ...param, default: git.ref || 'master' };
+      case 'IMAGE_NAME':
+        return { ...param, default: getImageUrl(name, namespace) };
+      default:
+        return param;
+    }
+  });
+
+  if (template.spec?.resources?.find((r) => r.type === 'git' && r.name === 'app-source')) {
     await createGitResource(git.url, namespace, git.ref);
+  }
+  if (template.spec?.resources?.find((r) => r.type === 'image' && r.name === 'app-image')) {
```

Comment:
```suggestion
  if (template.spec.resources?.find((r) => r.type === 'git' && r.name === 'app-source')) {
    await createGitResource(git.url, namespace, git.ref);
  }
  if (template.spec.resources?.find((r) => r.type === 'image' && r.name === 'app-image')) {
```
Spec is required in the type.

Reply:
I like this implementation that looks specifically for those names. It could blow back on us if they changed it upstream, but I think it's important that we try to aim for existing functionality. FWIW though, upstream recommends not to mix workspaces & pipeline resources... so this is likely overcautious code, but I see no downside really.

Reply:
Removed the optional check of spec. (And rebased)

Yeah I do not know if we want check here for just the type or the name as well. I do not know if all pipeline templates use exactly this names. Should we create a git resource if there is a resource with type === 'git' and name === 'app' ?! :thinking: 

If we want the "highest compatibly" I could remove the name check here... wdyt?

Reply:
Yeah, I was originally going to ask you to remove the name... but let us not. It's less intrusive if the user wants to use a git resource for something entirely different in their template. I think this implementation is fine 👍 .